tests/base.py

```diff
@@ -0,0 +1,25 @@
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License
+# 2.0; you may not use this file except in compliance with the Elastic License
+# 2.0.
+
+"""Shared resources for tests."""
+
+import unittest
+
+from detection_rules import rule_loader
+from detection_rules.rule import Rule
+
+
+class RuleLoadTest(unittest.TestCase):
+    """Base class for shared test cases which need to load rules"""
+
+    @classmethod
+    def setUpClass(cls):
+        cls.rule_files = rule_loader.load_rule_files(verbose=False)
+        cls.rules = rule_loader.load_rules(verbose=False)
```

Comment:
I always like to make `.rules` a list and `.rule_lookup` a dict. But both approaches are fine and have merits

Reply:
What the ...

```python
    @classmethod
    def setUpClass(cls):
        cls.rule_files = rule_loader.load_rule_files(verbose=False)
        cls.rule_lookup = rule_loader.load_rules(verbose=False)
        cls.rules = cls.rule_lookup.values()
        cls.production_rules = rule_loader.get_production_rules()
```

This causes a collision with the lark lib

<details>
<summary>stack</summary><br>


```
./env/detection-rules-build/bin/python -m detection_rules test
Traceback (most recent call last):
  File "/Users/jibarra/PycharmProjects/detection-rules-fork/env/detection-rules-build/lib/python3.7/site-packages/lark/load_grammar.py", line 800, in _get_parser
    return _get_parser.cache
AttributeError: 'function' object has no attribute 'cache'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/runpy.py", line 183, in _run_module_as_main
    mod_name, mod_spec, code = _get_module_details(mod_name, _Error)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/runpy.py", line 142, in _get_module_details
    return _get_module_details(pkg_main_name, error)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/runpy.py", line 109, in _get_module_details
    __import__(pkg_name)
  File "/Users/jibarra/PycharmProjects/detection-rules-fork/detection_rules/__init__.py", line 7, in <module>
    from . import devtools
  File "/Users/jibarra/PycharmProjects/detection-rules-fork/detection_rules/devtools.py", line 18, in <module>
    from eql import load_dump
  File "/Users/jibarra/PycharmProjects/detection-rules-fork/env/detection-rules-build/lib/python3.7/site-packages/eql/__init__.py", line 5, in <module>
    from .build import (
  File "/Users/jibarra/PycharmProjects/detection-rules-fork/env/detection-rules-build/lib/python3.7/site-packages/eql/build.py", line 5, in <module>
    from .engine import PythonEngine
  File "/Users/jibarra/PycharmProjects/detection-rules-fork/env/detection-rules-build/lib/python3.7/site-packages/eql/engine.py", line 11, in <module>
    from .transpilers import BaseEngine, BaseTranspiler
  File "/Users/jibarra/PycharmProjects/detection-rules-fork/env/detection-rules-build/lib/python3.7/site-packages/eql/transpilers.py", line 3, in <module>
    from .parser import parse_definitions, ignore_missing_functions
  File "/Users/jibarra/PycharmProjects/detection-rules-fork/env/detection-rules-build/lib/python3.7/site-packages/eql/parser.py", line 1167, in <module>
    'query_with_definitions', 'expr', 'signed_single_atom'])
  File "/Users/jibarra/PycharmProjects/detection-rules-fork/env/detection-rules-build/lib/python3.7/site-packages/lark/lark.py", line 276, in __init__
    self.grammar = load_grammar(grammar, self.source_path, self.options.import_paths, self.options.keep_all_tokens)
  File "/Users/jibarra/PycharmProjects/detection-rules-fork/env/detection-rules-build/lib/python3.7/site-packages/lark/load_grammar.py", line 1164, in load_grammar
    builder.load_grammar(grammar, source)
  File "/Users/jibarra/PycharmProjects/detection-rules-fork/env/detection-rules-build/lib/python3.7/site-packages/lark/load_grammar.py", line 1027, in load_grammar
    tree = _parse_grammar(grammar_text, grammar_name)
  File "/Users/jibarra/PycharmProjects/detection-rules-fork/env/detection-rules-build/lib/python3.7/site-packages/lark/load_grammar.py", line 841, in _parse_grammar
    tree = _get_parser().parse(text + '\n', start)
  File "/Users/jibarra/PycharmProjects/detection-rules-fork/env/detection-rules-build/lib/python3.7/site-packages/lark/load_grammar.py", line 813, in _get_parser
    _get_parser.cache = ParsingFrontend(lexer_conf, parser_conf, {})
  File "/Users/jibarra/PycharmProjects/detection-rules-fork/env/detection-rules-build/lib/python3.7/site-packages/lark/parser_frontends.py", line 80, in __init__
    self.parser = create_parser(lexer_conf, parser_conf, options)
  File "/Users/jibarra/PycharmProjects/detection-rules-fork/env/detection-rules-build/lib/python3.7/site-packages/lark/parser_frontends.py", line 163, in create_lalr_parser
    return LALR_Parser(parser_conf, debug=debug)
  File "/Users/jibarra/PycharmProjects/detection-rules-fork/env/detection-rules-build/lib/python3.7/site-packages/lark/parsers/lalr_parser.py", line 17, in __init__
    analysis = LALR_Analyzer(parser_conf, debug=debug)
  File "/Users/jibarra/PycharmProjects/detection-rules-fork/env/detection-rules-build/lib/python3.7/site-packages/lark/parsers/lalr_analysis.py", line 138, in __init__
    GrammarAnalyzer.__init__(self, parser_conf, debug)
  File "/Users/jibarra/PycharmProjects/detection-rules-fork/env/detection-rules-build/lib/python3.7/site-packages/lark/parsers/grammar_analysis.py", line 131, in __init__
    rules = parser_conf.rule_lookup + list(root_rules.values())
AttributeError: 'ParserConf' object has no attribute 'rule_lookup'
```
</details>

running `pytest test` directly doesn't result in the conflict since the rules module doesn't load

<details>
<summary>pytest</summary><br>

```
pytest tests
========================================================================================================================================================================= test session starts =========================================================================================================================================================================
platform darwin -- Python 3.7.4, pytest-5.4.3, py-1.9.0, pluggy-0.13.1
rootdir: /Users/jibarra/PycharmProjects/detection-rules-fork
collected 94 items                                                                                                                                                                                                                                                                                                                                                    

tests/test_all_rules.py ...................                                                                                                                                                                                                                                                                                                                     [ 20%]
tests/test_mappings.py ..                                                                                                                                                                                                                                                                                                                                       [ 22%]
tests/test_packages.py ......                                                                                                                                                                                                                                                                                                                                   [ 28%]
tests/test_schemas.py ....                                                                                                                                                                                                                                                                                                                                      [ 32%]
tests/test_toml_formatter.py ....                                                                                                                                                                                                                                                                                                                               [ 37%]
tests/test_utils.py ....                                                                                                                                                                                                                                                                                                                                        [ 41%]
tests/kuery/test_dsl.py .......                                                                                                                                                                                                                                                                                                                                 [ 48%]
tests/kuery/test_eql2kql.py ........                                                                                                                                                                                                                                                                                                                            [ 57%]
tests/kuery/test_evaluator.py .............                                                                                                                                                                                                                                                                                                                     [ 71%]
tests/kuery/test_kql2eql.py ..........                                                                                                                                                                                                                                                                                                                          [ 81%]
tests/kuery/test_lint.py ............                                                                                                                                                                                                                                                                                                                           [ 94%]
tests/kuery/test_parser.py .....     
```

</details>

Reply:
problem solved - some strange `env` dependency corruption. `make clean` resolved ... 🤷

Reply:
ad57c6ed9051488c996b35fdb3784b2655478ad4